apps/src/blockly/addons/cdoTrashcan.js

```diff
@@ -1,24 +1,167 @@
 import GoogleBlockly from 'blockly/core';
-import {ToolboxType} from '../constants';
 
-export default class CdoTrashcan extends GoogleBlockly.Trashcan {
+export default class CdoTrashcan extends GoogleBlockly.DeleteArea {
+  constructor(workspace) {
+    super();
+    this.workspace = workspace;
+    this.id = 'cdoTrashcan';
+
+    this.WIDTH_ = 47;
+    this.BODY_HEIGHT_ = 44;
+    this.LID_HEIGHT_ = 16;
+    this.MARGIN_TOP_ = 40;
+    this.SPRITE_LEFT_ = 0;
+    this.SPRITE_TOP_ = 32;
+    this.ANIMATION_LENGTH_ = 80;
+    this.ANIMATION_FRAMES_ = 4;
+    this.MAX_LID_ANGLE_ = 45;
+    this.OPACITY_MIN_ = 0.4;
+    this.OPACITY_MAX_ = 0.8;
+    this.TRASH_URL = '/blockly/media/trash.png';
+
+    this.isLidOpen = false;
+    this.lidTask_ = 0;
+    this.lidOpen_ = 0;
```

Comment:
This feels like a lot of magic numbers. Are there ways to minimize the number of magic numbers here?

Reply:
What if we added in comments to explain each value? 
https://github.com/code-dot-org/code-dot-org/commit/35ab00621a96099500eb15b5bcd07e53b8196134